lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -19,268 +17,38 @@ If TASK_SEARCH is not defined, it defaults to all tasks in the past 7 days and
-            'vendor/bootstrap/css/bootstrap.css',
-            'stylesheets/application.css'].each do |file|
-
-             filename = File.join(Gem::Specification.find_by_name("dynflow").gem_dir, 'web', 'assets', file)
-             FileUtils.copy_file(filename, File.join(tmp_dir, File.basename(file)))
-           end
-       end
-
-      def self.generate_index(tasks)
-        html = "<div><table class=\"table\">"
-        tasks.order("started_at desc").all.each do |task|
-          html << "<tr><td><a href=\"#{task.id}.html\">#{task.label}</a></td><td>#{task.started_at}</td>\
-                   <td>#{task.state}</td><td>#{task.result}</td></tr>"
-        end
-        html << "</table></div>"
-      end
-    end
-
-    if ENV['TASK_SEARCH'].nil? && ENV['TASK_DAYS'].nil?
+    if task.task_search.nil? && task.task_days.nil?
       filter = "started_at > \"#{7.days.ago.to_s(:db)}\" || " \
-        "(result != success && started_at > \"#{60.days.ago.to_s(:db)})\""
+               "(result != success && started_at > \"#{60.days.ago.to_s(:db)}\")"
     else
-      filter = ENV['TASK_SEARCH'] || ''
+      filter = task.task_search || ''
     end
 
-    if (days = ENV['TASK_DAYS'])
+    if (days = task.task_days)
       filter += " && " unless filter == ''
       filter += "started_at > \"#{days.to_i.days.ago.to_s(:db)}\""
     end
 
-    format = ENV['TASK_FORMAT'] || 'html'
-    export_filename = ENV['TASK_FILE'] || "/tmp/task-export-#{DateTime.now.to_i}.#{format == 'csv' ? 'csv' : 'tar.gz'}"
-
-    tasks = ForemanTasks::Task.search_for(filter)
-
-    puts _("Gathering #{tasks.count} tasks.")
-    if format == 'html'
-      Dir.mktmpdir('task-export') do |tmp_dir|
-        PageHelper.copy_assets(tmp_dir)
-
-
-        renderer = TaskRender.new
-        total = tasks.count
+    filter
+  end
 
-        tasks.each_with_index do |task, count|
-          File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') {|file| file.write(PageHelper.pagify(renderer.render_task(task)))}
-          puts "#{count + 1}/#{total}"
-          count += 1
-        end
+  def plans(world, filter)
+    puts "Retrieveing plans!"
+    ids = ForemanTasks::Task.search_for(filter).pluck(:external_id)
+    world.persistence.find_execution_plans(:filters => { :uuid => ids })
+  end
 
-        File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
+  task :export_tasks => :environment do
 
-        sh("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
-      end
-    elsif format == 'csv'
-      CSV.open(export_filename, 'wb') do |csv|
-        csv << ['id', 'state', 'type', 'label', 'result', 'parent_task_id', 'started_at', 'ended_at']
-        tasks.each do |task|
-          csv << [task.id, task.state, task.type, task.label, task.result,
-                  task.parent_task_id, task.started_at, task.ended_at]
-        end
-      end
+    export_task = ::Dynflow::Tasks::Export.new do |t|
```

Comment:
I think we talked about this and agreed on feeding the original rake task the necessary data instead of overriding parts of it. Fixed anyway